backend/extraction/parsing.py

```diff
@@ -49,9 +71,29 @@ def _guess_mimetype(file_bytes: bytes) -> str:
 
 def convert_binary_input_to_blob(data: BinaryIO) -> Blob:
     """Convert ingestion input to blob."""
+    file_size_in_mb = _get_file_size_in_mb(data)
+
+    if file_size_in_mb > MAX_FILE_SIZE:
+        raise HTTPException(
+            status_code=413,
+            detail=f"File size exceeds the maximum limit of {MAX_FILE_SIZE} MB.",
+        )
+
     file_data = data.read()
     mimetype = _guess_mimetype(file_data)
     file_name = data.name
+
+    if mimetype == "application/pdf":
+        number_of_pages = _get_pdf_page_count(file_data)
```

Comment:
Should we fail or extract from first few pages what do you think is more intuitive?

Reply:
(This is OK for now obviously)